### PR TITLE
anwickes/c01-aws02

### PR DESCRIPTION
**Pull request by anwickes**
# Description

Exercise (e.g. C01-E01): c01-aws02

Student Name: Adam Wickes

- Use the CLI to create a S3 bucket and upload a file to it.
- Access the file on the S3 bucket from any of the EC2 instances created in the previous exercise.

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [x] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-aws02/anwickes/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        },
+        "State": "associating"
+    }
+}
+```
+
+# SSH to EC2 instance and attempt to access S3 bucket contents
+
+> ssh -i anwickesPub.pem ec2-user@54.206.32.27
+> aws s3 ls anwickes-devopsacademyau-bucket
```
Comment (alanlima) [inline classes/01class/exercises/c01-aws02/anwickes/README.md:100]: Here it is missing the command to download the file into your instance. 😅

File: classes/01class/exercises/c01-aws02/anwickes/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+> aws s3 ls s3://anwickes-devopsacademyau-bucket
+
+```
+2020-07-12 16:50:06        171 index.html
+```
+
+# Create IAM role that grants S3 access and attach to EC2 instance
+
+- Create role
+> aws iam create-role --role-name anwickes-devopsacademyau-bucket-role --assume-role-policy-document file://anwickes-devopsacademyau-bucket-role.json
```
Comment (alanlima) [inline classes/01class/exercises/c01-aws02/anwickes/README.md:35]: Could you please add the JSON file used for this command? 🙂

File: classes/01class/exercises/c01-aws02/anwickes/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+    }
+}
+```
+
+- Create policy and attach it to role
+> aws iam put-role-policy --role-name anwickes-devopsacademyau-bucket-role --policy-name anwickes-devopsacademyau-bucket-ro-policy --policy-document file://anwickes-devopsacademyau-bucket-ro-policy.json
```
Comment (alanlima) [inline classes/01class/exercises/c01-aws02/anwickes/README.md:59]: Same file, missing the JSON file for the policy document

File: classes/01class/exercises/c01-aws02/anwickes/README.md (hunk 1 above)
Comment (anwickes) [reply classes/01class/exercises/c01-aws02/anwickes/README.md]: Good pick up Alan.. Have added. Cheers :)

File: classes/01class/exercises/c01-aws02/anwickes/README.md (hunk 2 above)
Comment (anwickes) [reply classes/01class/exercises/c01-aws02/anwickes/README.md]: Added.

File: classes/01class/exercises/c01-aws02/anwickes/README.md (hunk 3 above)
Comment (anwickes) [reply classes/01class/exercises/c01-aws02/anwickes/README.md]: Added.

**Pull request merged**